R/compare_df_types.R

```diff
@@ -0,0 +1,245 @@
+#' Generate a comparison of data.frames (or similar objects) that indicates if
+#' they will successfully bind together by rows.
+#'
+#' @details Due to the returned "column_name" column, no input data.frame may be
+#'   named "column_name".
+#'
+#' @param ... data.frames or similar objects (such as tibbles), or lists of
+#'   data.frames (or similar objects).  The values may optionally be named
+#'   arguments; if named, the output column will be the name; if not named, the
+#'   output columne will be the data.frame name (see examples section).
+#' @param return Should a summary of "all" columns be returned, only return
+#'   "match"ing columns, or only "mismatch"ing columns?
+#' @param bind_check What method of binding should be used to determine matches?
+#'   With "rbind", columns missing from a data.frame would be considered a
+#'   mismatch (as in \code{base::rbind()}; with "bind_rows", columns missing
+#'   from a data.frame would be considered a match (as in
+#'   \code{dplyr::bind_rows()}.
+#' @return A data.frame with a column named "column_name" with a value named
+#'   after the input data.frames' column names, and then one column per
+#'   data.frame (named after the input data.frame).  If more than one input
+#'   would have the same column name, the column naming is from
+#'   \code{base::merge()} and may differ from expected naming.  The rows within
```

Comment:
`compare_df_cols(iris, iris)` shouldn't fail.  My intent was that you could gently shoot yourself in the foot with something like `compare_df_cols(iris=data.frame(A=1), iris=data.frame(B=1))`.  It becomes more important when you have multiple lists and the lists may have the same names within.  I was trying to warn that it looks weird if you do `compare_df_cols(iris, iris, iris)` because the names will be "iris.x", "iris.y", and "iris" where the no-suffix version is the third input.

I reworked the documentation sentence there.